File: FrontEnd/electronics-shop-app/src/app/Product/Models/product.ts
import { Category } from "src/app/Category/Models/category";

export class Product {
    id: number;
    creationDate: Date;
    modificationDate: Date;
    createdByUserId: number;
    modifiedByUserId: number;
    name: string
    price: any;
    discount: number;
    twoPiecesDiscount: number;
    count: number;
    description: string
    categoryId: number;
    categoryName: Category;
}

File: BackEnd/ElectronicsShop/ElectronicsShop.Entities/Entities/Product.cs
using ElectronicsShop.Common.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;

namespace ElectronicsShop.Entities.Entities
{
    public class Product : IEntityIdentity<long>, IEntityDateTimeSignature, IEntityUserSignature
    {
        public long Id { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime? ModificationDate { get; set; }
        public long? CreatedByUserId { get; set; }
        public long? ModifiedByUserId { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public int Discount { get; set; }
        public int TwoPiecesDiscount { get; set; }
        public int Count { get; set; }
        public string Description { get; set; }
        public long CategoryId { get; set; }
        public virtual Category Category { get; set; }
    }
}
